pathspider/cmd/upload.py

```diff
@@ -0,0 +1,284 @@
+    c.setopt(c.WRITEDATA, buffer)
+    c.setopt(pycurl.HTTPHEADER, headers)
+    c.perform()
+    c.close()
+    #returns server answer
+    return buffer.getvalue().decode('iso-8859-1')
+
+def upload_metadata(filename):
+    '''
+    Uploads Metadata to PTO
+    Updates metadata with PTO added fields
+    returns url for measurment data as string
+    
+    :param filename: filename of metadata
+    :type filename: str
+    :param file: metadata
+    :type file: :class: json
+    :return: str -- url for data file or error
+    '''
+    
+    logger = logging.getLogger("uploader")
+    #forge link
+    link = BASELINK + os.path.basename(filename)
+    content_type = ["Content-type: application/json", "Authorization: APIKEY " + TOKEN]
+    answer = upload_file (link, content_type, filename)
+    logger.debug("uploaded metadata")
+    #try format answer to .json for extracting URL
+    try:
+        data = json.loads(answer)
+        return (data["__data"])
+    except:
+        logger.debug("unexpected answer. excepted .json instead: " + answer)
+        return "Error"
+
+def upload_data(url, filename, fn_metadata):
+    '''
+    Uploads datafile to campaign on PTO server
+    
+    :param url: name of campaign the data belongs to
+    :type url: str
+    :param filename: filename of file to compress
+    :type filename: str
+    '''
+
+    logger = logging.getLogger("uploader")
+    content_type = ["Content-type: application/bzip2", "Authorization: APIKEY " + TOKEN]
+    answer = upload_file(url, content_type, filename)
+    logger.debug("uploaded data")
+    try:
+        data = json.loads(answer)
+        logger.info("upload complete")
+        save_json(fn_metadata, data)
+        logger.info("saved metadata: " + fn_metadata)
+    except:
+        logger.info("upload failed")
+        logger.debug("unexpected answer. excepted .json instead: " + answer)
+    
+def upload_file(url, headers, filename):
+    '''
+    Uploads file to campaign on PTO server
+    
+    :param url: name of campaign the data belongs to
+    :type url: str
+    :param filename: filename of file to compress
+    :type filename: str
+    :param headers: additional http headers
+    :type headers: str
+    :return: str -- answer from server
+    '''
+
+    #Buffer for server answer
+    buffer = BytesIO()
+    #create pycurl object
+    c = pycurl.Curl()
+    #set curl options
+    # -X PUT
+    c.setopt(c.URL, url)
+    c.setopt(c.WRITEDATA, buffer)
+    # upload the contents of this file --data-binary @file
+    c.setopt(c.UPLOAD, 1)
+    file = open(filename, "rb")
+    c.setopt(c.READDATA, file)
+    # -H "Content-type: application/xxx"
+    # -H "Authorization: APIKEY $TOKEN"
+    c.setopt(pycurl.HTTPHEADER, headers)
+    c.perform()
+    c.close()
+    # File must be kept open while Curl object is using it
+    file.close()
+    #returns server answer
+    return buffer.getvalue().decode('iso-8859-1')
+
+def save_json(name, data):
+    #save json file
```

Comment:
Redundant comment. I'm not going to comment on all of these, please look for other places there are comments that there do not need to be.